arch/riscv32/core/fatal.c

```diff
@@ -199,8 +196,9 @@ FUNC_NORETURN void _Fault(const NANO_ESF *esf)
 	__asm__ volatile("csrr %0, mcause" : "=r" (mcause));
 
 	mcause &= SOC_MCAUSE_EXP_MASK;
-
+#ifdef CONFIG_PRINTK
 	printk("Exception cause %s (%d)\n", cause_str(mcause), (int)mcause);
```

Comment:
are you sure you need this? if CONFIG_PRINTK isn't defined, the printk() macro evaluates to nothing

Reply:
I got a build error when I tried building without printk(), not sure why if what you say is true.